### ISSUE TITLE
Crash on empty git repo

### ISSUE THREAD

**Issue by tommythorn** [zee-editor/zee#11]
```
$ cargo install zee
....
$ cargo new anynewproject
     Created binary (application) `anynewproject` package
$ zee anynewproject/src/main.rs 
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: -9, klass: 4, message: "reference 'refs/heads/master' not found" }', /home/tommy/.cargo.x86_64-linux/registry/src/github.com-1ecc6299db9ec823/zee-0.1.2/src/components/buffer.rs:529:67
               note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
                                                                                            Aborted (core dumped)
```

**Comment by mcobzarenco** [comment]
Thank you for checking out zee and for the bug report. I was carelessly unwrapping assuming there exist some refs in a repo, I've just pushed a fix to correctly handle empty repos.

> (Completely OT: I was disappointed to find Ctrl-t didn't transpose the two chars as in Emacs).

:+1: Absolutely, feeling at home with default Emacs bindings is a main goal of the project. It's just a consequence of not using C-t myself in Emacs, but learnt something new today! I will implement the change -- also hopding to systematically implement all basic text editing commands from Emacs + macros (with same default bindings) longer term

**Comment by tommythorn** [comment]
Awesome, thanks.  I have 35 years of emacs muscle memory and it’s jarring to be ripped out of the zone when suddenly things behave differently:)

Also OT: is it possible to reclaim the bottom two lines and only show them on demand (like when a prompt is needed)?

**Comment by tommythorn** [comment]
I apologize my Rust-fu is weak and I apparently don't know how to install from git.  Plain `cargo install --path .` doesn't work running it from within `core` gets me a binary that say
```
Error: X11ClipboardError: XCB connection error: ClosedParseErr

Caused by:
    0: XCB connection error: ClosedParseErr
    1: Connection closed, error during parsing display string
```
which surprised me a great deal.

**Comment by mcobzarenco** [comment]
> Plain cargo install --path . doesn't work and running it from within core gets me a binary that say

It's a cargo workspace, so you need to `cd` to `core` like you already do :+1: 


> `Error: X11ClipboardError: XCB connection error: ClosedParseErr`

Ah, so zee does try to integrate with the system's clipboard, cross platform, using [crossclip](https://github.com/mcobzarenco/crossclip).  This is gated under feature `system-clipboard` which is enabled by default though (for X11 on Linux you need extra dependencies `xorg-dev libxcb-shape0-dev libxcb-xfixes0-dev`). Could you try without that feature? 
```
cargo install --no-default-features --path .
```

Integrating with the system's clipboard on Linux X or Wayloand, Mac and Windows has been really painful so far..

**Comment by tommythorn** [comment]
Ok, that works.

Now when I exit Zee I get
```
                 diff:thread 'main' panicked at 'tried drawing while the app is exiting', /home/tommy/.cargo.aarch64-linux/registry/src/github.com-1ecc6299db9ec823/zi-0.2.0/src/app.rs:223:17                            master     0: 0  Top ▇
                                                                      note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
                           Aborted (core dumped)
```
This is 100% reproducible:
$ cd zee/core
$ zee src/undo.rs
^x ^c

Here's with a backtrace:
```
                 diff:thread 'main' panicked at 'tried drawing while the app is exiting', /home/tommy/.cargo.aarch64-linux/registry/src/github.com-1ecc6299db9ec823/zi-0.2.0/src/app.rs:223:17                            master     0: 0  Top ▇                                                                      stack backtrace:                                  
                                                                                         0: rust_begin_unwind
                                                                                                                          at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
                                                                                          1: core::panicking::panic_fmt
                                                                                                                                    at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
                                                                                                      2: zi_term::Crossterm<PainterT>::run_event_loop
                                3: zee::start_editor
                                                       4: zee::main
                                                                   note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
                                  Aborted (core dumped)
```

**Comment by tommythorn** [comment]
PS: I noticed that 'Ctrl-U' and 'Meta->' etc aren't working

**Comment by mcobzarenco** [comment]
Sorry for the delay in getting back to you on this, I've been away. Indeed, this was a bug in the underlying library [zi](https://github.com/mcobzarenco/zi) that I refactored out to deal with low level rendering to the terminal -- you can check out the commits if interested. master branch was a bit in flux migrating fully to zi. The recent interest spurred me in wrapping up that transition. I've also just published a new version 0.2 as it addresses most of the existing issues (as well as a range of other known problems). 

I've also changed cycling through the themes to be bound to `C-x C-t` to leave `C-t` swap characters like it does in Emacs. (and laid the foundations for making bindings configurable and commands extensible).

I am working on improving documentation, build instructions, creating a website and a tutorial. But once that's done, I'll priortise reproducing most of the core text editing bindings from Emacs.

**Comment by tommythorn** [comment]
Thanks, I’ll test some more and give feedback. Let’s close this now and I can reopen another as needed.
